testsuites/CBLTester/upgrade_tests/test_cbl_upgrade.py

```diff
@@ -0,0 +1,133 @@
+    base_liteserv_version = params_from_base_suite_setup["base_liteserv_version"]
+    upgraded_liteserv_version = params_from_base_suite_setup["upgraded_liteserv_version"]
+    liteserv_platform = params_from_base_suite_setup["liteserv_platform"]
+    liteserv_host = params_from_base_suite_setup["liteserv_host"]
+    upgrade_cbl_db_name = "upgarded_db"
+    base_url = params_from_base_suite_setup["base_url"]
+    encrypted_db = params_from_base_suite_setup["encrypted_db"]
+    db_password = params_from_base_suite_setup["db_password"]
+    sg_db = "db"
+    sg_admin_url = params_from_base_suite_setup["sg_admin_url"]
+    sg_blip_url = params_from_base_suite_setup["target_url"]
+    sg_version = params_from_base_suite_setup["sync_gateway_version"]
+    server_version = params_from_base_suite_setup["server_version"]
+    cluster_config = params_from_base_suite_setup["cluster_config"]
+    sg_config = params_from_base_suite_setup["sg_config"]
+    cbs_ip = params_from_base_suite_setup["cbs_ip"]
+    utils_obj = params_from_base_suite_setup["utils_obj"]
+    server_url = params_from_base_suite_setup["server_url"]
+
+    supported_base_liteserv = ["1.4", "2.0.0", "2.1.5", "2.5.0"]
+    db = Database(base_url)
+    if encrypted_db:
+        if base_liteserv_version < "2.1.5":
+            pytest.skip("Encyption is supported from 2.1.0 onwards."
+                        "{} doesn't have encrypted db upgrade support".format(base_liteserv_version))
+        db_config = db.configure(password=db_password)
+        db_prefix = "travel-sample-encrypted"
+    else:
+        db_config = db.configure()
+        db_prefix = "travel-sample"
+
+    temp_db = db.create("temp_db", db_config)
+    time.sleep(1)
+    new_db_path = db.getPath(temp_db)
+    delimiter = "/"
+    if liteserv_platform == "net-msft" or liteserv_platform == "net-uwp":
+        delimiter = "\\"
+    new_db_path = "{}".format(delimiter).join(new_db_path.split(delimiter)[:-2]) +\
+                  "{}{}.cblite2".format(delimiter, upgrade_cbl_db_name)
+    db.deleteDB(temp_db)
+
+    old_liteserv_db_name = ""
+    if base_liteserv_version in supported_base_liteserv:
+        old_liteserv_db_name = db_prefix + "-" + base_liteserv_version
+    else:
+        pytest.skip("Run test with one of supported base liteserv version - ".format(supported_base_liteserv))
+
+    if liteserv_platform == "android":
+        prebuilt_db_path = "/assets/{}.cblite2.zip".format(old_liteserv_db_name)
+    elif liteserv_platform == "xamarin-android":
+        prebuilt_db_path = "{}.cblite2.zip".format(old_liteserv_db_name)
+    else:
+        prebuilt_db_path = "Databases/{}.cblite2".format(old_liteserv_db_name)
+
+    log_info("Copying db of CBL-{} to CBL-{}".format(base_liteserv_version, upgraded_liteserv_version))
+    prebuilt_db_path = db.get_pre_built_db(prebuilt_db_path)
+    assert "Copied" == utils_obj.copy_files(prebuilt_db_path, new_db_path)
+    cbl_db = db.create(upgrade_cbl_db_name, db_config)
```

Comment:
will this not create new database?

Reply:
no, if db exist it just returns the object of existing db.